docs/changelog/3_x.rst

```diff
@@ -0,0 +1,517 @@
+
+.. code-block:: bash
+
+  $ edgedb project upgrade --to-testing -I my_instance
+
+**Hosted instances**
+
+To upgrade a remote (hosted) instance, we recommend the following
+dump-and-restore process.
+
+1. Spin up an empty 3.0 instance. You can use one of our :ref:`deployment
+guides <ref_guide_deployment>`, but you will need to modify some of the
+commands to use our testing channel and the beta release.
+
+    Under Debian/Ubuntu, when adding the EdgeDB package repository, use this
+    command instead:
+
+    .. code-block:: bash
+
+        echo deb [signed-by=/usr/local/share/keyrings/edgedb-keyring.gpg] \
+          https://packages.edgedb.com/apt \
+          $(grep "VERSION_CODENAME=" /etc/os-release | cut -d= -f2) testing \
+          | sudo tee /etc/apt/sources.list.d/edgedb.list
+
+    Use this command for installation under Debian/Ubuntu:
+
+    .. code-block:: bash
+
+        sudo apt-get update && sudo apt-get install edgedb-3
+
+    Under CentOS/RHEL, use this installation command:
+
+    .. code-block:: bash
+
+        sudo yum install edgedb-3
+
+    In any required ``systemctl`` commands, replace ``edgedb-server-2`` with
+    ``edgedb-server-3``.
+
+    Under any Docker setups, supply the ``3.0-beta.1`` tag.
+
+2. Take your application offline, then dump your v2.x database with the CLI
+
+   .. code-block:: bash
+
+     $ edgedb dump --dsn <old dsn> --all my_database.dump/
+
+   This will dump the schema and contents of your current database to a file
+   on your local disk called ``my_database.dump``. The file name isn't
+   important.
+
+3. Restore the empty v3.x instance from the dump
+
+   .. code-block:: bash
+
+     $ edgedb restore --all my_database.dump/ --dsn <new dsn>
+
+   Once the restore is complete, update your application to connect to the new
+   instance.
+
+This process will involve some downtime, specifically during steps 2 and 3.
+
+
+Client libraries
+----------------
+
+We've released new versions of our JavaScript and Python client libraries that
+support all 3.0 features and implement the updated protocol. These versions
+are backwards compatible with v1.x and v2.x instances, so we encourage all users to
+upgrade.
+
+.. list-table::
+
+  * - :ref:`TypeScript/JS <edgedb-js-intro>`
+    - ``edgedb@0.21.0``
+  * - :ref:`Python <edgedb-python-intro>`
+    - ``edgedb@0.24.0``
+  * - `Golang <https://www.edgedb.com/docs/clients/go/index>`_
+    - ``edgedb@0.12.0``
+  * - `Rust <https://github.com/edgedb/edgedb-rust>`_
+    - ``edgedb-tokio@0.3.0``
+  * - `.NET <https://github.com/quinchs/EdgeDB.Net>`_ (community-maintained)
+    - ``EdgeDB.Net.Driver@0.3.0``
+  * - `Elixir <https://github.com/nsidnev/edgedb-elixir>`_
+      (community-maintained)
+    - ``edgedb@0.4.0``
+
+New features
+============
+
+Simplified SDL syntax
+---------------------
+
+As part of our commitment to delivering the best developer experience in
+databases, we've made our schema definition language (or SDL) easier to use.
```

Comment:
> I kinda like this, but on the other hand, the other very important reason was us making this change to enable future syntax for declaring ad-hoc types in queries and functions. I'd mention that. Please also link this: https://github.com/edgedb/rfcs/blob/master/text/1022-freetypes.rst

Ah, so that's why! I had been wondering if there was a benefit besides a slicker syntax.